enable litmus test

cmhe opened the pull request.

cmhe commented:
@mlegner: I could not get the litmus acceptance tests running on my system, however it seems to work on github itself... So maybe this is even better?

cmhe commented:
@mlegner: Oh btw. a lot of this code is generated again from: https://github.com/puppetlabs/pdk-templates/tree/main/moduleroot/.github/workflows

cmhe commented:
This uses [honeycomb](https://ui.honeycomb.io) BTW, which is some sort of timing system... I just created an account for me there to get an api key. But maybe we should replace this api key with one managed by someone at siemens.

cmhe commented:
> This uses [honeycomb](https://ui.honeycomb.io) BTW, which is some sort of timing system... I just created an account for me there to get an api key. But maybe we should replace this api key with one managed by someone at siemens.

I enabled honeycomb support, because apparently disabling it will cause the scripts by the pdk-template to no longer work: https://github.com/puppetlabs/pdk-templates/issues/477

mlegner commented:
@cmhe I've just created a honeycomb account and a "siemens" team and invited you.

mlegner commented on .github/workflows/auto_release.yml:
```diff
@@ -0,0 +1,90 @@
+  workflow_dispatch:
+
+env:
+  HONEYCOMB_WRITEKEY: zEepjMgnVKilj943ApiKvH 
```
We probably shouldn't include keys into our source files.

I've created a new one and added it as an [action secret](https://github.com/siemens/puppet-pam_pkcs11/settings/secrets/actions).

cmhe replied:
Hmm... so I just followed the integration guide on this here. Puppet modules seems to just put the key there:

https://github.com/puppetlabs/puppetlabs-motd/blob/main/.github/workflows/pr_test.yml#L8
https://github.com/puppetlabs/puppetlabs-mysql/blob/main/.github/workflows/pr_test.yml#L8
https://github.com/puppetlabs/puppetlabs-apt/blob/main/.github/workflows/pr_test.yml#L8

So I assumed that this would be the way to go... I will have to look into how to get this variable from there into the github action while still not modifying the template: https://github.com/puppetlabs/pdk-templates/blob/main/moduleroot/.github/workflows/pr_test.yml.erb#L9

mlegner replied:
Maybe I'm missing or misunderstanding something. But can't I just take their API key and send arbitrary data to their datasets?

cmhe replied:
TBH, I have not looked into honeycomb enough to really understand what it possible there and exactly what it does and why it is useful. Currently I am also not really sure if I will ever look into the data it collects. I just wanted to be as close as possible to the official modules and what the PDK does in order to make maintenance as easy as possible. And if just deactivating it causes the PDK to generate broken gitlab action configs, than I have no choice but to enable it for now.

So if someone for some reason is uploading entire movies via that key to honeycomb and starts sharing it with others, be my guest I guess ;)

cmhe replied:
I am now using the github secret instead of putting it directly into the files.

cmhe replied:
So I also split up the changes into multiple separate commits. This will also make review easier, because it allows to see what files are generated and what I have manually changed.

cmhe commented:
> @cmhe I've just created a honeycomb account and a "siemens" team and invited you.

Thanks! I guess you should also create a new environment for this project: https://ui.honeycomb.io/siemens-35/environments just to have it separated. I cannot do this, because I am not the owner of this group.

mlegner commented:
> > @cmhe I've just created a honeycomb account and a "siemens" team and invited you.
> 
> Thanks! I guess you should also create a new environment for this project: https://ui.honeycomb.io/siemens-35/environments just to have it separated. I cannot do this, because I am not the owner of this group.

Yes, noticed that too. I created a `github.com/siemens/puppet-pam_pkcs11` environment now.

mlegner reviewed (changes requested):
As most of this is auto-generated, I don't have many comments here.

I agree with disabling all the old distributions but would suggest to put that change into a separate commit.

Otherwise just the issue with the "hardcoded" API key.

cmhe committed:
fix(metadata.json): remove unsupported platforms

Remove EOL versions of all Linux distros.

Newer versions or RedHat (>=8) no longer provide a `pam_pkcs11` package.

Gentoo is removed, because lack of acceptance testing possible there.

Signed-off-by: Claudius Heine <ch@denx.de>

cmhe committed:
fix: remove beaker nodesets

Beaker is no longer used for acceptance testing, remove beaker specific
configuration.

Signed-off-by: Claudius Heine <ch@denx.de>

cmhe committed:
fix: convert to litmus acceptance testing

Add litmus specific dependencies and convert spec_helper_acceptance.rb
and tests to litmus.

Signed-off-by: Claudius Heine <ch@denx.de>

cmhe committed:
fix: enable PDK template generation for github-actions files

Only `.sync.yml` is manually modified. Other files are generated by PDK
based on the modification of `.sync.yml`.

Enable the gitlab-action generation, which also contains execution of
the litmus acceptance test.

RedHat and SLES causes errors on provisioning which seems unrelated to
this repo so disable them for now.

Signed-off-by: Claudius Heine <ch@denx.de>

cmhe committed:
fix: remove `.gitlab/workflows/tests.yml`

`.gitlab/workflows/tests.yml` is no longer required, because its
function is now handled by the gitlab actions files generated by the
PDK.

Signed-off-by: Claudius Heine <ch@denx.de>

cmhe committed:
fix: add support for SLES 15 by reordering systemd checks

Put systemd as the default option in RedHat and Suse based distros,
because new distro versions are expected to use systemd. This will
future-proof the checks and also adds support for SLES 15.

Signed-off-by: Claudius Heine <ch@denx.de>
